### PR TITLE
Avoid duplicating events upon import

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/87654

During the work done for ticket 79975 I put in place the `array_map` code that tries to convert alphanumeric strings (UIDs) to ints; the consequence is the array is filled with `0` and `1` and EA Client will never spot an existing ID thus duplicating the events.

### CODE REVIEW

**Comment by lucatume** [comment]
To remove the duplicated Events, Venues and Organizers created by this the following code can be pasted into the theme `functions.php` file or in a must-use plugin file after The Events Calendar has been deactivated:

```php
// Remove duplicated imported events
global $wpdb;

$id_keys = array(
	'_FacebookID',
	'_EventFacebookID',
	'_EventMeetupID',
	'_uid',
	'_EventOriginalID',
	'_VenueFacebookID',
	'_VenueMeetupID',
	'_OrganizerFacebookID',
	'_OrganizerMeetupID',
);

$id_keys_interval = "'" . implode( "','", $id_keys ) . "'";

$post_ids_and_uids = $wpdb->get_results( "SELECT m.post_id, m.meta_value
		  FROM {$wpdb->postmeta} m 
		  JOIN {$wpdb->posts} p 
		  ON p.ID = m.post_id
		  WHERE m.meta_key in ({$id_keys_interval})
		  AND p.post_type = 'tribe_events'"
);

$batch_size         = 20;
$immediately_delete = true;

if ( ! empty( $post_ids_and_uids ) ) {
	$grouped = array();

	foreach ( $post_ids_and_uids as $post_id_and_uid ) {
		if ( ! isset( $grouped[ $post_id_and_uid->meta_value ] ) ) {
			$grouped[ $post_id_and_uid->meta_value ] = array( $post_id_and_uid->post_id );
		} else {
			$grouped[ $post_id_and_uid->meta_value ][] = $post_id_and_uid->post_id;
		}
	}

	$items = array_splice( $grouped, 0, $batch_size );

	foreach ( $items as $uid => $post_ids ) {
		// the smaller post ID is the original event
		sort( $post_ids );
		// remove the original event from the list
		array_shift( $post_ids );
		foreach ( $post_ids as $post_id ) {
			wp_delete_post( $post_id, $immediately_delete );
		}
	}
}
```

**Comment by barryhughes** [comment]
Worth adding a changelog entry

File: src/Tribe/Aggregator/Event.php [hunk 1]
```diff
@@ -189,7 +189,7 @@ public function get_existing_ids( $origin, $values ) {
 
 		$key = "_{$fields[ $origin ]['target']}";
 		$post_type = Tribe__Events__Main::POSTTYPE;
-		$interval = "('" . implode( "','", array_map( 'absint', $values ) ) . "')";
+		$interval = "('" . implode( "','", $values ) . "')";
```
Comment (barryhughes) [inline src/Tribe/Aggregator/Event.php]: What are your thoughts on keeping the `array_map()` but with `esc_sql` as the callback?

**Comment by lucatume** [comment]
Added changelog entry

File: src/Tribe/Aggregator/Event.php (hunk 1 above)
Comment (lucatume) [reply src/Tribe/Aggregator/Event.php]: I've added it, good idea.

**Pull request merged**